WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorView.swift

```diff
@@ -53,54 +32,64 @@ struct CustomFieldEditorView: View {
             VStack(alignment: .leading, spacing: Layout.sectionSpacing) {
                 // Key Input
                 VStack(alignment: .leading, spacing: Layout.subSectionsSpacing) {
-                    Text("Key") // todo-13493: set String to be translatable
+                    Text(Localization.keyLabel)
                         .foregroundColor(Color(.text))
                         .subheadlineStyle()
 
-                    TextField("Enter key", text: $key) // todo-13493: set String to be translatable
-                        .bodyStyle()
+                    TextField(Localization.keyPlaceholder, text: $key)
+                        .foregroundColor(Color(.text))
+                        .subheadlineStyle()
                         .padding(insets: Layout.inputInsets)
                         .background(Color(.listForeground(modal: false)))
                         .overlay(
                             RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(Color(.separator))
                         )
-                        .onChange(of: key) { _ in
-                            checkForModifications()
-                        }
                 }
 
                 // Value Input
                 VStack(alignment: .leading, spacing: Layout.subSectionsSpacing) {
                     HStack {
-                        Text("Value") // todo-13493: set String to be translatable
+                        Text(Localization.valueLabel)
                             .foregroundColor(Color(.text))
                             .subheadlineStyle()
 
                         Spacer()
 
                         if !isReadOnlyValue {
-                            Button(action: {
-                                showRichTextEditor = true
-                            }) {
-                                Text("Edit in Rich Text Editor") // todo-13493: set String to be translatable
-                                    .font(.footnote)
+                            Picker(Localization.editorPickerLabel, selection: $showRichTextEditor) {
+                                Text(Localization.editorPickerText).tag(false)
+                                Text(Localization.editorPickerHTML).tag(true)
                             }
-                            .buttonStyle(.plain)
-                            .foregroundColor(Color(uiColor: .accent))
+                            .pickerStyle(.segmented)
+                            .fixedSize()
                         }
                     }
 
-                    TextEditor(text: isReadOnlyValue ? .constant(value) : $value)
-                        .bodyStyle()
+                    if showRichTextEditor {
+                        AztecEditorView(initialValue: value,
+                                        onContentChanged: { content in
+                            DispatchQueue.main.async {
+                                value = content
+                            }
```

Comment:
❓ Could you explain the need for using `DispatchQueue.main.async` here?

Reply:
Ah, I added this in the commit message but it wasn't obvious:

> Also fixes "Modifying state during view update" in AztecEditorView

This fixes the same issue as in the 314a522870ec1f953109b9cd0d5350d165e9d9ff commit.

Reply:
That error hints an incorrect use of SwiftUI. Using `DispatchQueue.main.async` is just a workaround, and I'm unsure it actually fixes the problem.

Instead, try using `@Binding` only:

```Swift
struct AztecEditorView: UIViewControllerRepresentable {
    @Binding var content: String

    func makeUIViewController(context: Context) -> AztecEditorViewController {
        let controller = EditorFactory().customFieldRichTextEditor(initialValue: content)

        guard let aztecController = controller as? AztecEditorViewController else {
            fatalError("EditorFactory must return an AztecEditorViewController, but returned \(type(of: controller))")
        }

        aztecController.onContentChanged = { content in
            self.content = content
        }
        return aztecController
    }
}
```

Reply:
Thanks, that solution makes sense to me but it's showing the same issue:

![Screenshot 2024-09-25 at 21 54 23](https://github.com/user-attachments/assets/484e2821-897a-4072-bf36-3b9471b79410)

(The warning only appears when actually tapping the "HTML" switcher in the Simulator).

Reply:
This happens because `content` is updated immediately upon view creation. A workaround:

<img width="835" alt="Screenshot 2024-09-26 at 09 57 06" src="https://github.com/user-attachments/assets/f969c10a-b2be-4a91-920a-de004c6d3578">

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/13925/commits/60d5f4b2c8f34738b34c54cfbb4539e829a1828d